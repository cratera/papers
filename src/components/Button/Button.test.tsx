import { render } from '@testing-library/react-native'

import { Button, ButtonProps } from '.'

const setup = (props: ButtonProps) => {
  return render(<Button {...props} />)
}

describe('Button', () => {
  it('renders correctly with given props', async () => {
    const tree = setup({
      label: 'Button',
    }).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
