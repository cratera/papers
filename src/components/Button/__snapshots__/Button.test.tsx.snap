// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly with given props 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <Text>
    Button
  </Text>
</View>
`;
